src/Spago/Command/Fetch.purs

```diff
@@ -360,28 +360,75 @@ toAllDependencies = foldl (Map.unionWith (\l _ -> l)) Map.empty
 getGitPackageInLocalCache :: forall a. PackageName -> GitPackage -> Spago (Git.GitEnv a) Unit
 getGitPackageInLocalCache name package = do
   let localPackageLocation = Config.getPackageLocation name (GitPackage package)
-  tempDir <- mkTemp' (Just $ printJson Config.gitPackageCodec package)
-  logDebug $ "Cloning repo in " <> tempDir
-  Git.fetchRepo package tempDir >>= case _ of
-    Left err -> die err
-    Right _ -> do
+  existsGitCache <- FS.exists gitCacheLocation
+  { globallyCached, sourceDir } <-
+    if existsGitCache
+      then do
+        logInfo $ "Using global git cache in " <> gitCacheLocation
+        isBranch <- Git.isBranch { ref: package.ref, path: gitCacheLocation }
+        when isBranch do
+          fetchRes <- Git.fetchRepo package gitCacheLocation
+          case fetchRes of
+            Left err -> die err
+            Right _ -> pure unit
```

Comment:
If the package is in cache we just copy it over, no need to check if it's a branch since it can't be:
```suggestion
        logDebug $ "Using global git cache in " <> gitCacheLocation
```